observability/src/test/java/io/grpc/observability/interceptors/GcpLogHelperTest.java

```diff
@@ -0,0 +1,565 @@
+          base.toBuilder()
+              .clearPeerAddress()
+              .build());
+    }
+
+    // status description is null
+    {
+      sinkWriter.logTrailer(
+          seqId,
+          serviceName,
+          methodName,
+          statusDescription.getCode().toStatus(),
+          nonEmptyMetadata,
+          EventLogger.LOGGER_CLIENT,
+          rpcId,
+          peerAddress);
+      verify(sink).write(
+          base.toBuilder()
+              .clearStatusMessage()
+              .build());
+    }
+  }
+
+  @Test
+  public void logRpcMessage() throws Exception {
+    long seqId = 1;
+    String serviceName = "service";
+    String methodName = "method";
+    String rpcId = "d155e885-9587-4e77-81f7-3aa5a443d47f";
+
+    GrpcLogRecord.Builder builder = GrpcLogRecord.newBuilder()
+            .setTimestamp(timestamp)
+            .setSequenceId(seqId)
+            .setServiceName(serviceName)
+            .setMethodName(methodName)
+            .setEventType(EventType.GRPC_CALL_REQUEST_MESSAGE)
+            .setEventLogger(EventLogger.LOGGER_CLIENT)
+            .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+            .setRpcId(rpcId);
+    GrpcLogRecord base = builder.build();
+    // request message
+    {
+      sinkWriter.logRpcMessage(
+          seqId,
+          serviceName,
+          methodName,
+          EventType.GRPC_CALL_REQUEST_MESSAGE,
+          message,
+          EventLogger.LOGGER_CLIENT,
+          rpcId);
+      verify(sink).write(base);
+    }
+    // response message, logged on client
+    {
+      sinkWriter.logRpcMessage(
+          seqId,
+          serviceName,
+          methodName,
+          EventType.GRPC_CALL_RESPONSE_MESSAGE,
+          message,
+          EventLogger.LOGGER_CLIENT,
+          rpcId);
+      verify(sink).write(
+          base.toBuilder()
+              .setEventType(EventType.GRPC_CALL_RESPONSE_MESSAGE)
+              .build());
+    }
+    // request message, logged on server
+    {
+      sinkWriter.logRpcMessage(
+          seqId,
+          serviceName,
+          methodName,
+          EventType.GRPC_CALL_REQUEST_MESSAGE,
+          message,
+          EventLogger.LOGGER_SERVER,
+          rpcId);
+      verify(sink).write(
+          base.toBuilder()
+              .setEventLogger(EventLogger.LOGGER_SERVER)
+              .build());
+    }
+    // response message, logged on server
+    {
+      sinkWriter.logRpcMessage(
+          seqId,
+          serviceName,
+          methodName,
+          EventType.GRPC_CALL_RESPONSE_MESSAGE,
+          message,
+          EventLogger.LOGGER_SERVER,
+          rpcId);
+      verify(sink).write(
+          base.toBuilder()
+              .setEventType(EventType.GRPC_CALL_RESPONSE_MESSAGE)
+              .setEventLogger(EventLogger.LOGGER_SERVER)
+              .build());
+    }
+  }
+
+  @Test
+  public void getPeerAddressTest() {
+    assertNull(GcpLogHelper.getPeerAddress(Attributes.EMPTY));
+    assertSame(
+        peer,
```

Comment:
`peer` is initialized in `setUp` but used only here. It is best to initialize such things in the test unless they are used widely in the test file in multiple tests (which is not the case with `peer`).

Reply:
Updated